File: JAVA/java/src/RelativeSortArray.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class RelativeSortArray {
    public static int[] relativeSortArray(int[] arr1, int[] arr2) {
        int maxElement = Arrays.stream(arr1).max().orElse(0);
        int[] count = new int[maxElement + 1];

        // Count occurrences of each element
        for (int element : arr1) {
            count[element]++;
        }

        List<Integer> result = new ArrayList<>();
        // Add elements as per relative order
        for (int value : arr2) {
            while (count[value] > 0) {
                result.add(value);
                count[value]--;
            }
        }

        // Add remaining elements in ascending order
        for (int num = 0; num <= maxElement; num++) {
            while (count[num] > 0) {
                result.add(num);
                count[num]--;
            }
        }

        // Convert ArrayList to array
        return result.stream().mapToInt(Integer::intValue).toArray();
    }

    public static void main(String[] args) {
        System.out.println(relativeSortArray(new int[] {2,3,1,3,2,4,6,7,9,2,19},new int[] {2,1,4,3,9,6}));

    }
}

File: python/Relative Sort Array.py
def relativeSortArray( arr1, arr2):
    max_element = max(arr1)
    count = [0] * (max_element + 1)

    # Count occurrences of each element
    for element in arr1:
        count[element] += 1

    # Add elements as per relative order
    result = []
    for value in arr2:
        while count[value] > 0:
            result.append(value)
            count[value] -= 1

    # Add remaining elements in ascending order
    for num in range(max_element + 1):
        while count[num] > 0:
            result.append(num)
            count[num] -= 1

    return result
print(relativeSortArray([2,3,1,3,2,4,6,7,9,2,19], [2,1,4,3,9,6]))